CONTRIBUTING.md

```diff
@@ -29,43 +29,95 @@ integration tests that need a running synapse instance. These tests reside in
 [README](./testing/matrix-sdk-integration-testing/README.md) to easily set up a
 synapse for testing purposes.
 
-## Commit messages and PR title guidelines
 
-Ideally, a PR should have a *proper title*, with *atomic logical commits*, and each commit
-should have a *good commit message*.
+# Writing changelog entries
 
-An *atomic logical commit* is one that is ideally small, can be compiled in isolation, and passes
-tests. This is useful to make the review process easier (help your reviewer), but also when running
-bisections, helping identifying which commit introduced a regression.
+We aim to maintain clear and informative changelogs that accurately reflect the
+changes in our project. This guide will help you write useful changelog entries
+using git-cliff, which fetches changelog entries from commit messages.
 
-A *good commit message* should be composed of:
+## Commit Message Format
 
-- a prefix to indicate which area/feature is related by the commit
-- a short description that would give sufficient context for a reviewer to guess what the commit is
-  about.
+Commit messages should be formatted as Conventional Commits. In addition, some
+git trailers are supported and have special meaning (see below).
 
-Examples of commit messages that aren't so useful:
+### Conventional Commits
 
-- “add new method“
-- “enhance performance“
-- “fix receipts“
+Conventional Commits are structured as follows:
 
-Examples of good commit messages:
+```
+<type>(<scope>): <short summary>
+```
+
+The type of changes which will be included in changelogs is one of the following:
+
+    feat: A new feature
+    fix: A bug fix
+    doc: Documentation changes
+    refactor: Code refactoring
+    perf: Performance improvements
+    ci: Changes to CI configuration files and scripts
+
+The scope is optional and can specify the area of the codebase affected (e.g.,
+olm, cipher).
```

Comment:
Can we make the scope not optional, please? :pleading_face: This is really useful to me to know whether I can help with a review or not (since I don't partake in crypto reviews, most of the time).

Reply:
Oh yeah, +1

Reply:
That has happened in the past, and we've managed to figure it out, by either:

- not worrying too much about it, and specify a single scope that made the most sense
- generalizing to something wider, like "sdk" as the scope
- specifying multiples scopes, e.g. `refactor(timeline, event_cache): rename DaThing`

Reply:
Oh man, multiple scopes would take up a lot of space and people already complain that conventional commits take up too much space. Not sure if I like that idea.

> I can help with a review or not (since I don't partake in crypto reviews, most of the time).

Would code owners be a better thing to polish? We already have them but the core SDK team gets assigned to all reviews:

https://github.com/matrix-org/matrix-rust-sdk/blob/9c6413551cd6e15f01453e57178fe98e0d12200b/.github/CODEOWNERS#L2

Perhaps leaving me for the crypto crate and the removing the `rust` team from it would make more sense?

Reply:
That would work too. I considered adding myself as a code owner of the timeline at some point, maybe something we can chat about in other venues.

Reply:
Alright, let's then leave the scope optional and tweak the `CODEOWNERS` to solve your problem.